src/pages/home/report/ReportActionItemFragment.js

```diff
@@ -152,21 +157,32 @@ function ReportActionItemFragment(props) {
                 </Text>
             );
         }
-        case 'TEXT':
-            return (
+        case 'TEXT': {
+            const textFragment = (
+                <Text
+                    numberOfLines={props.isSingleLine ? 1 : undefined}
+                    style={[
+                        styles.chatItemMessageHeaderSender,
```

Comment:
I don't think we really want to reuse the same `<Text>` component and props here as  `styles.chatItemMessageHeaderSender` isn't appropriate for the system message. I also think we need to have some handling for RTL as in the `COMMENT` case:

https://github.com/Expensify/App/blob/cf006036f4e8e480027b9e5c0b9b847bfd41f10d/src/pages/home/report/ReportActionItemFragment.js#L127-L138

Otherwise we end up with this issue:

![image](https://github.com/Expensify/App/assets/27287420/78dae9ae-ee60-4b47-9edd-8b8ef72b7e4a)

Reply:
Some of the styles in `styles.chatItemMessageHeaderSender` apply to the system message like the `fontSize`, `lineHeight`, etc. 

Here's how the text "marked as complete" in a task is styled when a task is marked complete. 

https://github.com/Expensify/App/blob/e1083781e4b98fa4c2c9cf49cf1742b7d89fa2ba/src/components/ReportActionItem/TaskAction.js#L24

Should we follow the same approach for the `<Text>` component that will be meant for system messages now that you prefer having two separate ones?

Are we also going to not use `props.isSingleLine` for said component?

Reply:
My point is that you needed to add an extra style to override those in `styles.chatItemMessageHeaderSender`, which doesn't really make sense when we can just no use it here (since it's not a header). Seems like we could take inspiration from `TaskAction`, yes, and also for the `COMMENT` fragments (to handle RTL).

Reply:
@jjcoffee This is what I'm getting so far.  

> I don't think we really want to reuse the same <Text> component

1. Since we're not reusing the component, there'll be two `Text` components - one for system messages and another that'll be wrapped in a tooltip as usual. The `style` prop for the system message will be similar to the one in `TaskAction` so that we're not overwriting `styles.chatItemMessageHeaderSender`, but the remaining props will be similar to the `Text` wrapped in a tooltip.

2. The `Text` component for system messages needs to handle RTL like regular messages do (I'd also like to know how you got the text to start from right to left for testing).

3. Since the system message will have its own `Text` component styled using existing styles like in `TaskAction`, the helper method will no longer be necessary and will be removed.

Does all this sound right 👍?

Reply:
Sounds good! You can get RTL by entering e.g. an arabic display name.

Reply:
@jjcoffee Does this look right to you for an Arabic display name?

![expensify-OD-system-message-with-arabic-name](https://github.com/Expensify/App/assets/79470910/03ba0476-db2c-4bfb-bc4c-9da93dd46c9d)

Reply:
I think the bracket is meant to come after

Reply:
How about now?

![expensify-OD-correct-rtl-format](https://github.com/Expensify/App/assets/79470910/9483bc10-aa99-45c4-8266-4cf610181424)

Reply:
Yup looks good! Btw you can verify on other system messages, e.g. inviting a user with an Arabic display name to a workspace.

Reply:
Good to know. Still testing on other platforms.

Reply:
Just iOS left to test with my new solution. 

Ran into a few hiccups, but I'm back on the right track.

Reply:
I'm grateful you had [concerns](https://github.com/Expensify/App/pull/30193#issuecomment-1790397963) about indiscriminately wrapping the fragments in `Text` because I discovered this would've caused a regression on Android native where really long messages overflowed the screen's bounds. 

This wasn't happening earlier, or maybe I just didn't notice, but either way, I found a way to conditionally wrap the fragments in `Text` so we have less to worry about in terms of regressions. 

All this is done in this commit f669405e5edda8668d0c4dbfd1d2e56c37622f91.

I also used separate `Text` components in `ReportActionItemFragment` so we're not overwriting styles, added styling to account for RTL scenarios, and got rid of the helper method in `StyleUtils` since it's no longer needed. 

Everything tests well, and I hope it's the same on your end. 

Hopefully, it's not EOD for you yet, and we can still hit the soft [deadline](https://github.com/Expensify/App/issues/27544#issuecomment-1790406897) you set earlier, but if things still feel amiss, feel free to let me know and we can tighten things up even further.